### ISSUE TITLE
Improve logging with systemd integration

### ISSUE THREAD

**Issue by hekmon** [rclone/rclone#4596]
#### What is your current rclone version (output from `rclone version`)?
rclone v1.53.1
- os/arch: linux/amd64
- go version: go1.15

#### What problem are you are trying to solve?
As rclone can already be aware of systemd presence thru the notify socket (used by the mount command with systemd type notify) it would be nice to enhance a bit the systemd integration:

- **Cleaner logs**
The logger flags are not adjusted when running under systemd: it overloads the logs like this:
  ```
  Sep 15 05:36:00 server rclone[2838611]: 2020/09/15 05:36:00 INFO  : some/file: vfs cache: removed cache file as file deleted
  ```
  Date and time are present 2 times: one from journald and one from the rclone logger. Because rclone is already able to detect if it is launched by systemd it would be nice to change the rclone logger default flags when systemd launch is detected to remove the rlcone log date and time in order to have cleaner logs.

- **Prettier (and better integrated) logs**
journald handles [priority levels](https://wiki.archlinux.org/index.php/Systemd/Journal#Priority_level) just fine and can adapt output with text formating and colors for each level to facilitate log parsing by humans. It also allows an operator to filter any log/priority level using `journalctl` even if they are fired by the software and saved by journald for example.
There is [several ways](http://0pointer.de/blog/projects/journal-submit.html) to do that but in that case the easiest way is to prefix each line with a balise like `<0>`. A moddest logger implementation using this feature can be found [here](https://github.com/hekmon/hllogger). 
I am aware rclone does have its own more sophisticated logger: this would mean adding an auto prefixing feature if systemd launch is detected.

**Comment by ncw** [comment]
This seems like a good idea . I'm sure we could adjust rclone's logger very easily.

Do you think the notify signal is enough? Potentially you can run of the services other than mount eg rclone serve off systemd too.

Do you want to have a go at this?

**Comment by hekmon** [comment]
You are right. While having the notify socket annouced thru env means there is 99,9% of chances we are launched by systemd, the feature itself should be activatable manually if needed for other (futures ?) cases.

So this would mean:

- a way to activate systemd support (a flag ?)
- automatically turning this feature on if the [notify socket is detected](https://github.com/iguanesolutions/go-systemd/blob/v4.0.1/notify.go#L12) ?

I can definitly take a swig at it even if I not really familliar with the rlcone codebase (I won't be able to give an ETA :D)

**Comment by ncw** [comment]
> So this would mean:
> 
> * a way to activate systemd support (a flag ?)
> * automatically turning this feature on if the [notify socket is detected](https://github.com/iguanesolutions/go-systemd/blob/v4.0.1/notify.go#L12) ?

Sounds good! I'd suggest --log-systemd or something like that.



> I can definitly take a swing at it even if I not really familliar with the rlcone codebase (I won't be able to give an ETA :D)

Great.

Let me know if you'd like some pointers on where to start.

**Comment by hekmon** [comment]
#4609 ready for review :)

**Comment by hekmon** [comment]
Fixed by #4609
